packages/katello/katello/hostname-change.rb

```diff
@@ -551,7 +551,9 @@ def run
           STDOUT.puts result
           STDOUT.puts 'Restarting puppet services'
           run_cmd('/bin/systemctl try-restart puppet')
-          run_cmd('foreman-maintain service restart --only puppetserver')
+          if File.exist?('/usr/lib/systemd/system/puppetserver.service')
```

Comment:
This felt cleaner than shelling out to `rpm -q puppetserver`, but still not ideal.

We also could parse answers, but…

Reply:
How about `systemctl is-active puppetserver --quiet` and checking the exit code? From the man page:
> Check whether any of the specified units are active (i.e. running). Returns an exit code 0 if at least one is active, or non-zero otherwise.

Reply:
done